config/Config-images.in

```diff
@@ -16,6 +16,7 @@ menu "Target Images"
 			prompt "Compression"
 			default TARGET_INITRAMFS_COMPRESSION_LZMA if TARGET_apm821xx
 			default TARGET_INITRAMFS_COMPRESSION_LZMA if TARGET_ar71xx
+			default TARGET_INITRAMFS_COMPRESSION_LZMA if TARGET_ath79
```

Comment:
Will this affect other devices not using LZMA?

Reply:
I honestly don't know. In any case, it should only affect the initramfs images, not the sysupgrade ones. It was there for ar71xx but was not added to ath79, but this is a weak argument.

Is there any way to make this compression per-device?

Reply:
What happens if you do not add this line here?

Reply:
Based on the discussion in IRC, we _think_ that this line is not required, as a similar effect will be done by the "lzma" in `KERNEL_INITRAMFS := kernel-bin | append-dtb | lzma | loader-kernel`.
So, I will drop this line in my staging tree. Let's see whether it still works then.

Reply:
> What happens if you do not add this line here?

The board fails to decompress the kernel when booting the initramfs image via TFTP:
```
OpenWrt kernel loader for AR7XXX/AR9XXX
Copyright (C) 2011 Gabor Juhos <juhosg@openwrt.org>
Decompressing kernel... rimary_ap
```
and stays there. While, when the line is added, it gets to boot correctly:
```
OpenWrt kernel loader for AR7XXX/AR9XXX
Copyright (C) 2011 Gabor Juhos <juhosg@openwrt.org>
Decompressing kernel... done!
Starting kernel at 80060000...

[    0.000000] Linux version 4.19.97...
```

> Based on the discussion in IRC, we think that this line is not required

You're absolutely right.

Reply:
Your last line contradicts the reports before. So, is `default TARGET_INITRAMFS_COMPRESSION_LZMA if TARGET_ath79` required to use initramfs or not?

Reply:
19:48 PaulFertser: Also, his feedback kinda suggests there's a bug in "OpenWrt kernel loader". But it might also mean that he's loading to some inapproriate address and so when decompressing a kernel with uncompressed initramfs the "big" initramfs overwrites something in that loader.
19:49 PaulFertser: So it's like parts of loader are overwritten by the uncompressed big (since initramfs is not compressed) kernel image.

Reply:
Therefore, either the `default TARGET_INITRAMFS_COMPRESSION_LZMA if TARGET_ath79` line is added, or LZMA compression is manually selected during compilation, or the initramfs image can't be booted.

By the way, adding `KERNEL_INITRAMFS := kernel-bin | append-dtb | lzma | loader-kernel` to the .mk file does nothing, so it seems the line is useless anyway.

Reply:
Sorry for the misunderstanding. So, *not* adding the TARGET_INITRAMFS_COMPRESSION_LZMA line makes the initramfs not bootable (`Decompressing kernel... rimary_ap`). Adding it makes the initramfs image bootable.

Reply:
> 19:48 PaulFertser: Also, his feedback kinda suggests there's a bug in "OpenWrt kernel loader". But it might also mean that he's loading to some inapproriate address and so when decompressing a kernel with uncompressed initramfs the "big" initramfs overwrites something in that loader.
> 19:49 PaulFertser: So it's like parts of loader are overwritten by the uncompressed big (since initramfs is not compressed) kernel image.
> 19:50 blogic: lyma compression fails tnd to be offset issues
> 19:50 blogic: lzma decompression fails tnd to be offset issues
> 19:51 blogic: so the ram needs to hold the lzma, the decompressor and the final image
> 19:51 blogic: if the decompress overrides the origianl image or decompress code it'll fail
> 19:51 blogic: this is normally the root cause of the bug he talks about
> 19:52 blogic: the lzma-loader uses the stock lzma lib code so chances are slim its a loader bug

Sadly, I don't have any output from the bootloader, the first text I can read is `OpenWrt kernel loader for AR7XXX/AR9XXX`. All possible interaction is pressing the reset button. :disappointed:

Reply:
TL;DR: The uncompressed kernel is too big and overwrites the lzma loader (which is at a defined position):
https://github.com/openwrt/openwrt/blob/master/target/linux/ath79/image/Makefile#L13

Try to move the lzma loader by putting a bigger value for LZMA_TEXT_START, e.g.
0x81000000 (=16 MB offset)
0x81500000 (=24 MB offset)

Then we expect it to work without adding the line in Config-images.in

I'm just about to check for the optimum value in parallel.

IRC:
19:53 PaulFertser: Yes. So it's more likely that the lzma loader is located at a bad place. 80A00000 is just ~10 megabytes away from kernel image start.
19:57 PaulFertser: adrianschmutzler: I'd suggest changing LZMA_TEXT_START=0x80a00000 in image/Makefile to something more appropriate so that a bigger uncompressed initramfs would fit without overwriting the loader.
linmob (~quassel@85.93.13.231) has quit IRC: Quit: No Ping reply in 180 seconds.
20:03 adrianschmutzler: PaulFertser: what would be a reasonable value? What are the boundary conditions for choosing?
20:03 adrianschmutzler: And would I change this for all devices then, or make it variable?
20:04 PaulFertser: adrianschmutzler: I think it should be closer to the end of the RAM. So assuming all devices have at least 32 MiB it can be about that minus the loader size.
linmob (~quassel@85.93.13.231) has joined #openwrt-devel
20:05 adrianschmutzler: Ah, I was thinking the wrong way. So this value determines the start of the loader and the uncompressed image is BEFORE
20:06 adrianschmutzler: and thus it overwrites the loader if too big
20:07 adrianschmutzler: so, how do I determine the size of the loader?
20:10 PaulFertser: adrianschmutzler: ./staging_dir/toolchain-mips_24kc_gcc-8.3.0_musl/bin/mips-openwrt-linux-size $(find -name loader.elf)
20:10 PaulFertser: (I do not have it built right atm)

Reply:
`LZMA_TEXT_START=0x81000000` works, and also does `LZMA_TEXT_START=0x81500000`. Initramfs is now booted without the TARGET_INITRAMFS_COMPRESSION_LZMA setting.